support/Environments/sonic.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/env sh
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+spectre_setup_modules() {
+    echo "All modules on Sonic are provided by the system"
+}
+
+spectre_load_modules() {
+    # The order here is important
+    module load use.own
+    # module load gcc/11.1.0
+    module load spectre-env > /dev/null 2>&1
+}
+
+spectre_unload_modules() {
+    # The order here is important
+    module unload spectre-env > /dev/null 2>&1
+    # module unload gcc/11.1.0
+    module unload use.own
+}
+
+spectre_run_cmake_gcc_release() {
+    if [ -z ${SPECTRE_SOURCE} ]; then
+        echo "You must set SPECTRE_SOURCE to the cloned SpECTRE directory"
+        return 1
+    fi
+
+    if [ -z ${SPECTRE_INSTALL} ]; then
+        echo "You must set SPECTRE_INSTALL to the target directory"
+        return 1
```

Comment:
We typically don't require this to be set. You should just run `spectre_run_cmake` directly in your build directory. Is there a particular reason you'd like this to be set on Sonic?

Reply:
I would like to keep the build directory separate from the directory from which SpECTRE is run on Sonic. This, e.g. avoids creation of multiple build directories, one for each type of build, cluttering with the build files, etc.

Please let me know if you have any concerns.

Reply:
> Sorry, there’s something I’m not understanding about the workflow you would like to use and what we are currently doing. Currently our workflow on all clusters is
> 
> 1. Clone spectre into `SPECTRE_HOME` and then choose a build directory anywhere else, doesn’t have to be inside the source tree.
> 2. Do `spectre_run_cmake` inside the build directory, which builds against the `SPECTRE_HOME` source tree and all output goes into the build directory.
> 3. Then, when we run an executable, we typically copy the executable to the run directory, which can be anywhere and doesn’t have to be inside the build directory (we typically don’t run in the build directory!).
> 
> Now, I think we have support for an install directory, but this CMake feature is really geared at end-users who don’t want to develop or change anything in the code. This essentially adds the `make install` step to the build process. For most use cases, I don’t envision this being the right way to go, since inevitably (at least right now) some sort of debugging and code development will need to happen.

>3
As I understand, this is this default workflow when submitting jobs to queues. Does SpECTRE also copy executables to another location auomatically when we are testing code by directing calling the executables, say e.g. in a jupyter notebook? This can be seen e.g. in [example notebooks](https://github.com/sxs-collaboration/spectre/blob/develop/docs/Examples/BbhInitialData/BbhInitialData.ipynb) in SpECTRE. I find myself doing this for debugging using pythonic interface.